In [2]:
%run 01_cascade_model_baseline_vs_improved.ipynb


Baseline ingots per puller: 111.81
Improved ingots per puller: 129.9
Puller amplification ratio: 1.162
Baseline wafers per ingot: 13018.0
Improved wafers per ingot: 17039.0
Wafer amplification ratio: 1.309
Baseline wafers/year: 978,131,850
Improved wafers/year: 1,487,423,920

Total system amplification ratio: 1.521
Baseline capacity (GW): 9.96
Improved capacity (GW): 15.14
Capacity amplification ratio: 1.521
Absolute GW gain: 5.18
Pullers required at baseline efficiency: 1022
Current pullers: 672
Pullers avoided due to yield improvements: 350
Pullers avoided: 350
Capex avoided (USD): 104,968,753
Capex avoided (Million USD): 104.97
Energy per ingot (kWh): 6960
Annual ingots avoided: 39123
Annual energy avoided (MWh): 272298
Annual energy avoided (GWh): 272.3
Annual electricity cost avoided (INR): 1,906,084,295
Annual electricity cost avoided (Crore INR): 190.61


In [3]:
# ---------------------------
# Pullers required without improvements
# ---------------------------

required_pullers_baseline = (gw_2 / gw_1) * pullers

print("Pullers required at baseline efficiency:",
      round(required_pullers_baseline))

print("Current pullers:", pullers)

pullers_avoided = required_pullers_baseline - pullers

print("Pullers avoided due to yield improvements:",
      round(pullers_avoided))


Pullers required at baseline efficiency: 1022
Current pullers: 672
Pullers avoided due to yield improvements: 350


In [4]:
# ---------------------------
# Capex Avoidance Calculation
# ---------------------------

capex_per_puller = 300000  # USD

capex_avoided = pullers_avoided * capex_per_puller

print("Pullers avoided:", round(pullers_avoided))
print("Capex avoided (USD):", format(int(capex_avoided), ","))
print("Capex avoided (Million USD):",
      round(capex_avoided / 1e6, 2))

Pullers avoided: 350
Capex avoided (USD): 104,968,753
Capex avoided (Million USD): 104.97


In [5]:
# ---------------------------
# Energy Avoidance Calculation
# ---------------------------

# Energy per ingot (kWh)
energy_per_ingot = (200 * 18) + (70 * 48)

# Baseline ingots per puller
ingots_baseline = ingots_1

# Total ingots avoided per year
ingots_avoided_per_year = pullers_avoided * ingots_baseline

# Total energy avoided per year
energy_avoided_kwh = ingots_avoided_per_year * energy_per_ingot

print("Energy per ingot (kWh):", energy_per_ingot)
print("Annual ingots avoided:", round(ingots_avoided_per_year))
print("Annual energy avoided (MWh):",
      round(energy_avoided_kwh / 1000))
print("Annual energy avoided (GWh):",
      round(energy_avoided_kwh / 1e6, 2))


Energy per ingot (kWh): 6960
Annual ingots avoided: 39123
Annual energy avoided (MWh): 272298
Annual energy avoided (GWh): 272.3


In [6]:
# ---------------------------
# Annual Electricity Cost Avoided
# ---------------------------

cost_per_kwh_inr = 7  # INR per kWh

annual_cost_avoided_inr = energy_avoided_kwh * cost_per_kwh_inr

print("Annual electricity cost avoided (INR):",
      format(int(annual_cost_avoided_inr), ","))

print("Annual electricity cost avoided (Crore INR):",
      round(annual_cost_avoided_inr / 1e7, 2))


Annual electricity cost avoided (INR): 1,906,084,295
Annual electricity cost avoided (Crore INR): 190.61


## Executive Impact Summary

Process improvements (puller yield + cycle time + wafer thickness + kerf + yield cascade) resulted in:

• Pullers avoided: ~350 units  
• Direct Capex avoided: ~$105M  
• Annual energy avoided: ~272 GWh  
• Annual electricity cost avoided: ~₹190 Crore  

Strategic Insight:
Yield improvement and wafer process optimization provide capital-equivalent expansion 
without physical infrastructure growth.

This demonstrates how upstream engineering improvements cascade into 
large-scale economic and operational impact.
